send/tinia

```diff
@@ -1,119 +1,35 @@
-#!/bin/bash
+#!/usr/bin/env python3
+import subprocess
 
-SERVICE_NAME="tinia"
+import send_lib
+import sys
+import re
 
-FACILITY_NAME=$1
-DESTINATION=$2
-DESTINATION_TYPE=$3
+service_name = "tinia"
 
-if [ -z "$DESTINATION" ]; then
-	echo "Missing Destination argument (DB NAME there)" >&2
-	exit 231
-fi
+send_lib.check_input_fields(sys.argv, destination_type_required=True)
 
-if [ -z "$FACILITY_NAME" ]; then
-	echo "Missing FacilityName argument" >&2
-	exit 232
-fi
+facility = sys.argv[1]
+destination = sys.argv[2]
+destination_type = sys.argv[3]
 
-if [ -z "$DESTINATION_TYPE" ]; then
-	echo "Destination type of this service can't be empty" >&2
-	exit 233;
-else
-	TYPE="service-specific"
-	if [ "$DESTINATION_TYPE" != "$TYPE" ]; then
-		echo "Destination type of this service need to be $TYPE" >&2
-		exit 234;
-	fi
-fi
+send_lib.check_destination_type_allowed(destination_type, "service-specific")
+send_lib.check_destination_format(destination, destination_type, send_lib.SIMPLE_PATTERN)
 
 #Destination is name of database on oracle in localhost tnsnames file
-DBNAME=$DESTINATION
+dbname = destination
 
-SERVICE_FILES_BASE_DIR="`pwd`/../gen/spool"
-SERVICE_FILES_DIR="$SERVICE_FILES_BASE_DIR/$FACILITY_NAME/$SERVICE_NAME"
+service_files_dir = send_lib.get_gen_folder(facility, service_name)
+send_lib.create_lock(service_name, dbname)
 
-#Just safety check. This should not happen.
-if [ ! -d "$SERVICE_FILES_DIR" ]; then echo '$SERVICE_FILES_DIR: '$SERVICE_FILES_DIR' is not a directory' >&2 ; exit 1; fi
+#  copy service file to temporary directory
+with send_lib.prepare_temporary_directory() as tmp_dir:
+	send_lib.copy_files_to_directory(service_files_dir, tmp_dir, re.compile(service_name))
 
-#Create lock (same like in slave)
-LOCK_DIR=${LOCK_DIR:=/var/lock}
-LOCK_FILE="${LOCK_DIR}/perunv3-${SERVICE_NAME}-$DBNAME.lock"
-LOCK_PIDFILE="$LOCK_FILE/pid"
+execscript = "./tinia_process.pl"
+send_lib.check_script_file(execscript)
+command = [execscript, "-d", dbname, "-p", tmp_dir, "-s", service_name]
+process = subprocess.Popen(command, stdout=subprocess.PIPE)
```

Comment:
just an idea - couldn't we move all of this logic to send_lib and create a function like 'exec_custom_script(execscript, arguments)'? Skimmed through rest of the send scripts and it seems that this appears in multiple send scripts (insight_mu for example), so I guess we would save a couple lines of code